liblangutil/ParserBase.cpp

```diff
@@ -57,30 +57,78 @@ Token ParserBase::advance()
 	return m_scanner->next();
 }
 
+string ParserBase::tokenName(Token _token)
+{
+	if (_token == Token::Identifier)
+		return "identifier";
+	else if (_token == Token::EOS)
+		return "end of source";
+	else if (TokenTraits::isReservedKeyword(_token))
+		return "reserved keyword '" + TokenTraits::friendlyName(_token) + "'";
+	else if (TokenTraits::isElementaryTypeName(_token)) //for the sake of accuracy in reporting
+	{
+		ElementaryTypeNameToken elemTypeName = m_scanner->currentElementaryTypeNameToken();
+		return "'" + elemTypeName.toString() + "'";
+	}
+	else
+		return "'" + TokenTraits::friendlyName(_token) + "'";
+};
+
 void ParserBase::expectToken(Token _value, bool _advance)
 {
 	Token tok = m_scanner->currentToken();
 	if (tok != _value)
 	{
-		auto tokenName = [this](Token _token)
+		string const expectedToken = ParserBase::tokenName(_value);
+		if (m_parserErrorRecovery)
+			parserError("Expected " + expectedToken + " but got " + tokenName(tok));
+		else
+			fatalParserError("Expected " + expectedToken + " but got " + tokenName(tok));
+		// Do not advance so that recovery can sync or make use of the current token. This is especially useful if the expected token
```

Comment:
Could add a line break after the first sentence.

Reply:
Seems fixed in af86f4274d

Reply:
(And again, thanks!)